addons/source-python/packages/source-python/players/_base.py

```diff
@@ -221,20 +217,36 @@ def is_fake_client(self):
         """
         return self.playerinfo.is_fake_client()
 
-    def is_hltv(self):
+    @cached_property
+    def _is_hltv(self):
```

Comment:
Why is this wrapper needed?

Reply:
This would break compatibility, because they are currently not properties but functions. So existing codes are doing `pl.is_<bot/hltv>()` and not having them wrapped would means they would now get a `bool isn't callable` exception.

Reply:
Ah, I didn't thought about that. Thanks for the clarification.